orbit/pkg/update/notifications_test.go

```
@@ -181,79 +205,103 @@ func TestWindowsMDMEnrollmentPrevented(t *testing.T) {
-	cfg, err := enrollFetcher.GetConfig()
-	assertResult(cfg, err)
-
-	// unblock the first call and wait for it to complete
-	close(chProceed)
-	time.Sleep(100 * time.Millisecond)
-
-	// this next call won't execute the command because of the frequency
-	// restriction (it got called less than N seconds ago)
-	cfg, err = enrollFetcher.GetConfig()
-	assertResult(cfg, err)
-
-	// wait for the fetcher's frequency to pass
-	time.Sleep(enrollFetcher.Frequency)
-
-	// this call executes the command, and it returns the Is Windows Server error
-	apiErr = errIsWindowsServer
-	cfg, err = enrollFetcher.GetConfig()
-	assertResult(cfg, err)
-
-	// this next call won't execute the command (both due to frequency and the
-	// detection of windows server)
-	cfg, err = enrollFetcher.GetConfig()
-	assertResult(cfg, err)
-
-	// wait for the fetcher's frequency to pass
-	time.Sleep(enrollFetcher.Frequency)
+			var (
+				apiCallCount int
+				apiErr       error
+			)
+			chProceed := make(chan struct{})
+			fetcher := &windowsMDMEnrollmentConfigFetcher{
+				Fetcher:   baseFetcher,
+				Frequency: 2 * time.Second, // just to be safe with slow environments (CI)
+			}
+			if cfg.NeedsProgrammaticWindowsMDMEnrollment {
+				fetcher.execEnrollFn = func(args WindowsMDMEnrollmentArgs) error {
+					<-chProceed    // will be unblocked only when allowed
+					apiCallCount++ // no need for sync, single-threaded call of this func is guaranteed by the fetcher's mutex
+					return apiErr
+				}
+				fetcher.execUnenrollFn = func(args WindowsMDMEnrollmentArgs) error {
+					panic("should not be called")
+				}
+			} else {
+				fetcher.execUnenrollFn = func(args WindowsMDMEnrollmentArgs) error {
+					<-chProceed    // will be unblocked only when allowed
+					apiCallCount++ // no need for sync, single-threaded call of this func is guaranteed by the fetcher's mutex
+					return apiErr
+				}
+				fetcher.execEnrollFn = func(args WindowsMDMEnrollmentArgs) error {
+					panic("should not be called")
```

question: Is it ok to call panic() here instead of logging an error and recovering from this condition?

---

Yes, this is a test and a panic will fail the test.

---

thanks, I was going back on forth on the code and totally missed the context of this being a test